components/server/src/auth/verification-service.ts

```diff
@@ -59,6 +60,14 @@ export class VerificationService {
         if (!this.verifyService) {
             throw new Error("No verification service configured.");
         }
+        const isBlockedNumber = this.userDB.isBlockedPhoneNumber(phoneNumber);
+        const usages = await this.userDB.countUsagesOfPhoneNumber(phoneNumber);
+        if (usages > 3) {
+            throw new Error("The given phone number has been used more than three times.");
+        }
+        if (await isBlockedNumber) {
+            throw new Error("The given phone number is blocked due to abuse.");
```

Comment:
Can we throw this as a user error here? Otherwise it will count towards server errors in our metrics.